src/Evolution/Ringdown/StrahlkorperCoefsInRingdownDistortedFrame.cpp

```diff
@@ -0,0 +1,96 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+#include "Evolution/Ringdown/StrahlkorperCoefsInRingdownDistortedFrame.hpp"
+
+#include "DataStructures/Matrix.hpp"
+#include "Domain/CoordinateMaps/Distribution.hpp"
+#include "Domain/Creators/Sphere.hpp"
+#include "Domain/Creators/SphereTimeDependentMaps.hpp"
+#include "Domain/StrahlkorperTransformations.hpp"
+#include "IO/H5/Dat.hpp"
+#include "IO/H5/File.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/IO/ReadSurfaceYlm.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/Strahlkorper.hpp"
+#include "Utilities/Gsl.hpp"
+
+#include <array>
+#include <cstddef>
+#include <optional>
+#include <vector>
+
+namespace evolution::Ringdown {
+std::vector<DataVector> strahlkorper_coefs_in_ringdown_distorted_frame(
+    const std::string& path_to_horizons_h5,
+    const std::string& surface_subfile_name,
+    const size_t requested_number_of_times_from_end, const double match_time,
+    const double settling_timescale,
+    const std::array<double, 3>& exp_func_and_2_derivs,
+    const std::array<double, 3>& exp_outer_bdry_func_and_2_derivs,
+    const std::array<std::array<double, 4>, 3>& rot_func_and_2_derivs) {
+  // Read the AhC coefficients from the H5 file
+  const std::vector<ylm::Strahlkorper<Frame::Inertial>>& ahc_inertial_h5 =
+      ylm::read_surface_ylm<Frame::Inertial>(
+          path_to_horizons_h5, surface_subfile_name,
+          requested_number_of_times_from_end);
+    const size_t l_max{ahc_inertial_h5[0].l_max()};
+
+  // Read the AhC times from the H5 file
+  h5::H5File<h5::AccessType::ReadOnly> ahc_h5_file(path_to_horizons_h5);
+  const Matrix& coefs_for_times =
+      ahc_h5_file.get<h5::Dat>(surface_subfile_name).get_data();
+  std::vector<double> ahc_times{};
+  for (size_t i = 0; i < coefs_for_times.rows(); ++i) {
+    ahc_times.push_back(coefs_for_times(i, 0));
+  }
+  ahc_h5_file.close();
+
+  // Create a time-dependent domain; only the the time-dependent map options
+  // matter; the domain is just a spherical shell with inner and outer
+  // radii chosen so any conceivable common horizon will fit between them.
+  const domain::creators::sphere::TimeDependentMapOptions::ShapeMapOptions
+      shape_map_options{
+          l_max, domain::creators::time_dependent_options::
+                     KerrSchildFromBoyerLindquist{1.0, {{0.0, 0.0, 0.0}}}};
```

Comment:
If you want a spherical map (i.e. the identity for shape), pass `std::nullopt` instead of `KerrSchildFromBoyerLindquist`